src/ssl.c

```diff
@@ -20484,6 +20531,186 @@ void wolfSSL_sk_GENERAL_NAME_free(WOLFSSL_STACK* sk)
 #endif /* OPENSSL_EXTRA || WOLFSSL_WPAS_SMALL */
 
 #ifdef OPENSSL_EXTRA
+
+static void wolfSSL_DIST_POINT_NAME_free(WOLFSSL_DIST_POINT_NAME* dpn)
+{
+    if (dpn != NULL) {
+        if (dpn->name.fullname != NULL) {
+            wolfSSL_GENERAL_NAMES_free(dpn->name.fullname);
+        }
+        XFREE(dpn, NULL, DYNAMIC_TYPE_OPENSSL);
+    }
+}
+
+
+/* returns new pointer on success and NULL on fail */
+static WOLFSSL_DIST_POINT_NAME* wolfSSL_DIST_POINT_NAME_new(void)
+{
+    WOLFSSL_DIST_POINT_NAME* dpn = NULL;
+    WOLFSSL_GENERAL_NAMES* gns = NULL;
+
+    dpn = (WOLFSSL_DIST_POINT_NAME*)XMALLOC(sizeof(WOLFSSL_DIST_POINT_NAME),
+                                            NULL, DYNAMIC_TYPE_OPENSSL);
+    if (dpn == NULL) {
+        return NULL;
+    }
+    XMEMSET(dpn, 0, sizeof(WOLFSSL_DIST_POINT_NAME));
+
+    gns = wolfSSL_sk_new_null();
+    if (gns == NULL) {
+        WOLFSSL_MSG("wolfSSL_sk_new_null error");
+        XFREE(dpn, NULL, DYNAMIC_TYPE_OPENSSL);
+        return NULL;
+    }
+    gns->type = STACK_TYPE_GEN_NAME;
+
+    dpn->name.fullname = gns;
+    dpn->type = CRL_DIST_OID;
+
+    return dpn;
+}
+
+
+/* Creates and returns new DIST_POINT structure */
+WOLFSSL_DIST_POINT* wolfSSL_DIST_POINT_new(void)
+{
+    WOLFSSL_DIST_POINT* dp = NULL;
+    WOLFSSL_DIST_POINT_NAME* dpn = NULL;
+
+    WOLFSSL_ENTER("DIST_POINT_new");
+
+    dp = (WOLFSSL_DIST_POINT*)XMALLOC(sizeof(WOLFSSL_DIST_POINT), NULL,
+                                      DYNAMIC_TYPE_OPENSSL);
+    if (dp == NULL) {
+        return NULL;
+    }
+    XMEMSET(dp, 0, sizeof(WOLFSSL_DIST_POINT));
+
+    dpn = wolfSSL_DIST_POINT_NAME_new();
+    if (dpn == NULL) {
+        XFREE(dp, NULL, DYNAMIC_TYPE_OPENSSL);
+        return NULL;
+    }
+    dp->distpoint = dpn;
+
+    return dp;
+}
+
+
+/* Frees DIST_POINT objects.
+*/
+void wolfSSL_DIST_POINT_free(WOLFSSL_DIST_POINT* dp)
+{
+    WOLFSSL_ENTER("wolfSSL_DIST_POINT_free");
+    if (dp != NULL) {
+        wolfSSL_DIST_POINT_NAME_free(dp->distpoint);
+        dp->distpoint = NULL;
```

Comment:
Here you `NULL` the member but in `wolfSSL_DIST_POINT_NAME_free` you don't. I think we can skip this `NULL` step.